sdk/identity/identity/README.md

```diff
@@ -0,0 +1,31 @@
+## Azure Identity client library for Javascript
```

Comment:
I've added this library to the exclusion list in `.docsettings.yml` and linked it to the overarching issue #1583 since it seems there's a README sweep planned.  However, I definitely want to get good README contents written before then, just probably in a different PR after this initial one is merged.

Is it OK if I file an issue to track that work to be done next week after the rest of `@azure/identity` is written?

Reply:
Another issue to track the final look of README is absolutely fine.
As part of this PR, I would suggest to remove the template and have just one section on how you expect the users to use this library. A few lines that covers the crux and one or two code snippets is enough

Reply:
Sounds good, I'll add that!

Reply:
I added a couple of hypothetical examples for how this would be used with `@azure/keyvault-keys` and some details on how a SDK library author would integrate it.  Let me know what you think!